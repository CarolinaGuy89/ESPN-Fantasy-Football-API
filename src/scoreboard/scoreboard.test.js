import ApiModel from '../api-model/api-model.js';
import Scoreboard from './scoreboard.js';

import { localObject, serverResponse } from './scoreboard.stubs.js';

describe('Scoreboard', () => {
  let scoreboard;

  beforeEach(() => {
    scoreboard = new Scoreboard();
  });

  afterEach(() => {
    scoreboard = null;
  });

  test('extends ApiModel', () => {
    expect(scoreboard).toBeInstanceOf(ApiModel);
  });

  describe('attribute population from server response', () => {
    beforeEach(() => {
      scoreboard = Scoreboard.buildFromServer(serverResponse);
    });

    test('parses data correctly', () => {
      expect(scoreboard).toMatchSnapshot();
    });
  });

  describe('attribute population from local object', () => {
    beforeEach(() => {
      scoreboard = Scoreboard.buildFromLocal(localObject);
    });

    test('parses data correctly', () => {
      expect(scoreboard).toMatchSnapshot();
    });
  });
});
