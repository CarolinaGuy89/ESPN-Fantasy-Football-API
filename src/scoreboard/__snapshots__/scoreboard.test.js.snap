// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard attribute population from local object parses data correctly 1`] = `
Scoreboard {
  "dateOfFirstNFLGameInScoringPeriod": "2017-11-10T01:25:00.000Z",
  "leagueId": undefined,
  "matchupPeriodId": 10,
  "nflGamesInProgress": false,
  "scoringPeriodId": 10,
  "seasonId": undefined,
}
`;

exports[`Scoreboard attribute population from server response parses data correctly 1`] = `
Scoreboard {
  "dateOfFirstNFLGameInScoringPeriod": "2017-11-10T01:25:00.000Z",
  "leagueId": undefined,
  "matchupPeriodId": 10,
  "nflGamesInProgress": false,
  "scoringPeriodId": 10,
  "seasonId": undefined,
}
`;
